### PR TITLE
test: Migrate from mocha to vitest

### PR DESCRIPTION
**Pull request by jimmycallin**
- [x] I have added automatic tests where applicable
- [x] The PR title is suitable as a release note
- [x] The PR contains a description of what has been changed
- [x] The description contains manual test instructions

## Changes

Migrating from mocha to vitest for testing to reduce dependencies.

We are already using vite for building, and vitest has a similar API to jest while being much faster and requiring fewer dependencies.

## Test

- Run the tests, are they working?
- Compare session.test.js to previous test_session.js. It should be similar with small updates due to callback -> async/await migration.

### CODE REVIEW

File: vitest.setup.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { fetch } from "cross-fetch";
```
[the remark this answers is not in the document]
Comment (gismya) [reply vitest.setup.js:1]: Not needed if we are targeting node 18, which we now are according to the nvmrc.

**Review by gismya — CHANGES_REQUESTED** [review]
The old `test_sesson.js` should probably be removed.

File: vitest.setup.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jimmycallin) [reply vitest.setup.js:1]: unfortunately msw cannot intercept native fetch yet, so we need to overwrite it with cross-fetch https://github.com/mswjs/interceptors/issues/159#issuecomment-1227468245

**Review by gismya — APPROVED** [review]
Very good. Love that the tests now work in Node 18.

**Pull request merged**